### ISSUE TITLE
Multi stage build fails when referencing a previous stage with ARG

### ISSUE THREAD

**Issue by erandagan** [GoogleContainerTools/kaniko#747]
**TLDR** `FROM <previous-stage-alias>-${SOME_ARG}`  fails Kaniko.

This issue presents a special case of #233 where the stage alias contains a reference to an `ARG`, a pattern supported by docker & BuildKit.

**Actual behavior**
A handy feature of Multistage builds is the ability to dynamically select a base stage in later stages:

```Dockerfile
ARG VARIANT

FROM scratch AS base
# Shared stuff over here...

FROM base AS base-dev
# Some dev only stuff here, like installing debugging tools.

FROM base AS base-prod
# Some prod only stuff here, like running Release builds or installing prod-only dependencies.

FROM base-${VARIANT}
# Final image dynamically selected. 
# Some final processing steps here, like ENTRYPOINT.
```
In the above example, when `VARIANT` is set to dev, the final image will inherit from `base-dev`. This pattern works great on stock Docker & BuildKit (which even skips unused branches, like `base-prod`, to speed up the build further). However, on Kaniko, the above build will fail

`VARIANT=dev`:
```
INFO[0003] Downloading base image base-dev              
2019/08/22 11:32:08 No matching credentials were found, falling back on anonymous
error building image: UNAUTHORIZED: authentication required; [map[Name:library/base-dev Action:pull Type:repository Class:]]
```
But, if we replace `FROM base-${VARIANT}` with a hardcoded `FROM base-dev`, everything is resolved correctly, though it renders the dynamic base picking useless.

So, it seems that Kaniko doesn't properly resolve stage aliases when they contain an ARG expression.

**Expected behavior**
Kaniko should correctly handle `FROM <previous-stage-name>-${SOME_ARG}` instructions.

**To Reproduce**
1. Write a Dockerfile similar to the one above
2. Attempt to build it with Kaniko (should fail)
3. Attempt to build it using `docker` (should succeed)

**Additional Information**
 - Dockerfile
    As presented in the example above.
 - Build Context
    `--build-arg VARIANT=dev`
 - Kaniko Image (fully qualified with digest)
    `gcr.io/kaniko-project/executor@sha256:78d44ec4e9cb5545d7f85c1924695c89503ded86a59f92c7ae658afa3cff5400`

**Comment by tejal29** [comment]
@erandagan Currently we do not support evaluating args.
We do support resolving env varaibles in the stage name 
https://github.com/GoogleContainerTools/kaniko/blob/master/pkg/dockerfile/dockerfile.go#L67

We need to do something similar something similar for Args.

Feel free to take a stab at it.
